include/seqan3/argument_parser/validators.hpp

```diff
@@ -451,6 +455,23 @@ class file_validator_base
             return detail::to_string(" Valid file extensions are: [", extensions | views::join(std::string{", "}), "].");
     }
 
+    /*!\brief Helper function that checks if a string is a suffix of another string. Case insensitive.
+     * \param str The string to be searched.
+     * \param suffix The suffix to be searched for.
+     * \returns `true` if `suffix` is a suffix of `str`, otherwise `false`.
+     */
+    bool string_ends_with(std::string_view str, std::string_view suffix) const
+    {
+        size_t const suffix_length{suffix.size()};
+        size_t const str_length{str.size()};
+        return suffix_length > str_length ? 
+               false : 
+               std::ranges::equal(str.substr(str_length - suffix_length), suffix, [] (char const chr1, char const chr2)           
+               {
+                   return std::tolower(chr1) == std::tolower(chr2);
+               });
```

Comment:
Can we use the definition from https://en.cppreference.com/w/cpp/string/basic_string_view/ends_with ?

```c++
size() >= sv.size() && compare(size() - sv.size(), npos, sv) == 0
```

One range less :)